rinad/src/ipcm/ipcm.cc

```diff
@@ -1151,6 +1154,55 @@ IPCManager_::select_policy_set(Addon* callee, Promise* promise,
 	return IPCM_PENDING;
 }
 
+// Returns IPCM_SUCCESS if a kernel plugin was successfully loaded/unloaded,
+// IPCM_FAILURE otherwise
+ipcm_res_t
+IPCManager_::plugin_load_kernel(const std::string& plugin_name,
+				bool load)
+{
+	ostringstream ss;
+	ipcm_res_t result = IPCM_FAILURE;
+	pid_t pid;
+
+	pid = fork();
+	if (pid < 0) {
+		// parent, fork() failed
+		ss << "Kernel plugin (un)loading: fork() failed";
+		FLUSH_LOG(ERR, ss);
+		result = IPCM_FAILURE;
+
+	} else if (pid == 0) {
+		// child
+		if (load) {
+			execl("modprobe", "modprobe", plugin_name.c_str(),
+			      NULL);
+			execlp("modprobe", "modprobe", plugin_name.c_str(),
```

Comment:
Why 2 calls to exec? execlp should be enough I guess

(Commented also in the other PR, sorry for duplicating)